### ISSUE TITLE
should aggregation be more robust to errors in files?

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#1901]
#### What is the problem?

The entire aggregation of metadata (the new version) would blow up if e.g. some file could not be loaded correctly.  I think,
- it should be sufficient to issue a warning and skip that file.
- the entire aggregation for a (sub)dataset should also be guarded to catch any exception thrown, and then if possible also provide the same mechanism as `--on-failure` done with other commands (or just issue a warning and proceed)

currently:
```shell
(git)smaug:/mnt/btrfs/scrap/datalad/tmp/datasets/openfmri/ds000009[master]
$> datalad --dbg -l debug aggregate-metadata -r
[DEBUG  ] Determined class of decorated function: <class 'datalad.metadata.aggregate.AggregateMetaData'> 
[DEBUG  ] Resolved dataset for metadata aggregation: <Dataset path=/mnt/btrfs/scrap/datalad/tmp/datasets/openfmri/ds000009> 
[DEBUG  ] Determined class of decorated function: <class 'datalad.interface.annotate_paths.AnnotatePaths'> 
[DEBUG  ] Determined class of decorated function: <class 'datalad.distribution.subdatasets.Subdatasets'> 
[DEBUG  ] Resolved dataset for subdataset reporting/modification: <Dataset path=/mnt/btrfs/scrap/datalad/tmp/datasets/openfmri> 
[INFO   ] Aggregate metadata for dataset /mnt/btrfs/scrap/datalad/tmp/datasets/openfmri/ds000009 
[DEBUG  ] Determined class of decorated function: <class 'datalad.distribution.subdatasets.Subdatasets'> 
[DEBUG  ] Resolved dataset for subdataset reporting/modification: <Dataset path=/mnt/btrfs/scrap/datalad/tmp/datasets/openfmri/ds000009> 
[DEBUG  ] Dump metadata of <Dataset path=/mnt/btrfs/scrap/datalad/tmp/datasets/openfmri/ds000009> (merge mode: init) into <Dataset path=/mnt/btrfs/scrap/datalad/tmp/datasets/openfmri/ds000009> 
[DEBUG  ] Metadata parser unusable: audio 
[DEBUG  ] Metadata parser unusable: exif 
[DEBUG  ] Metadata parser unusable: xmp 
Traceback (most recent call last):
  File "/home/yoh/proj/datalad/datalad/venvs/dev/bin/datalad", line 8, in <module>
    main()
  File "/home/yoh/proj/datalad/datalad/datalad/cmdline/main.py", line 347, in main
    ret = cmdlineargs.func(cmdlineargs)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/base.py", line 427, in call_from_parser
    ret = list(ret)
  File "/home/yoh/proj/datalad/datalad/datalad/interface/utils.py", line 382, in generator_func
    result_renderer, result_xfm, _result_filter, **_kwargs):
  File "/home/yoh/proj/datalad/datalad/datalad/interface/utils.py", line 449, in _process_results
    for res in results:
  File "/home/yoh/proj/datalad/datalad/datalad/metadata/aggregate.py", line 671, in __call__
    to_save)
  File "/home/yoh/proj/datalad/datalad/datalad/metadata/aggregate.py", line 199, in _extract_metadata
    paths=relevant_paths)
  File "/home/yoh/proj/datalad/datalad/datalad/metadata/metadata.py", line 639, in _get_metadata
    for loc, meta in contentmeta_t or {}:
  File "/home/yoh/proj/datalad/datalad/datalad/metadata/parsers/base.py", line 37, in <genexpr>
    ((k, v) for k, v in self._get_content_metadata()) if content else None
  File "/home/yoh/proj/datalad/datalad/datalad/metadata/parsers/bids.py", line 117, in _get_content_metadata
    for r in yield_participant_info(participants_fname):
  File "/home/yoh/proj/datalad/datalad/datalad/metadata/parsers/bids.py", line 130, in yield_participant_info
    for row in csv.DictReader(tsvfile, dialect=dialect):
  File "/usr/lib/python2.7/csv.py", line 108, in next
    row = self.reader.next()
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2019' in position 72: ordinal not in range(128)
()
> /usr/lib/python2.7/csv.py(108)next()
-> row = self.reader.next()
(Pdb) 

```
and
```shell
$> datalad --on-failure ignore aggregate-metadata -r      
[INFO   ] Aggregate metadata for dataset /mnt/btrfs/scrap/datalad/tmp/datasets/openfmri/ds000009 
[ERROR  ] 'ascii' codec can't encode character u'\u2019' in position 72: ordinal not in range(128) [csv.py:next:108] (UnicodeEncodeError) 
(dev)2 12067 ->1
```

**Comment by yarikoptic** [comment]
pushed a "Fix" for the first in 5f2b0364595fae9e1b8a42bc22a6e295bee9019a

**Comment by mih** [comment]
I think this is good now.
